core/chains/evm/headtracker/head_listener.go

```diff
@@ -28,32 +32,63 @@ var (
 	}, []string{"evmChainID"})
 )
 
-type headListener struct {
-	config           Config
-	ethClient        evmclient.Client
+type headListener[
+	HTH htrktypes.Head[BLOCK_HASH, ID],
+	S commontypes.Subscription,
+	ID txmgrtypes.ID,
+	BLOCK_HASH commontypes.Hashable,
+] struct {
+	config           htrktypes.Config
+	client           htrktypes.Client[HTH, S, ID, BLOCK_HASH]
 	logger           logger.Logger
 	chStop           utils.StopChan
-	chHeaders        chan *evmtypes.Head
-	headSubscription ethereum.Subscription
+	chHeaders        chan HTH
+	headSubscription commontypes.Subscription
 	connected        atomic.Bool
 	receivingHeads   atomic.Bool
 }
 
-// NewHeadListener creates a new HeadListener
-func NewHeadListener(lggr logger.Logger, ethClient evmclient.Client, config Config, chStop chan struct{}) httypes.HeadListener {
-	return &headListener{
-		config:    config,
-		ethClient: ethClient,
-		logger:    lggr.Named("HeadListener"),
-		chStop:    chStop,
+type evmHeadListener = headListener[*evmtypes.Head, ethereum.Subscription, *big.Int, common.Hash]
+
+var _ commontypes.HeadListener[*evmtypes.Head, common.Hash] = &evmHeadListener{}
+
+func NewHeadListener[
+	H commontypes.Head[BLOCK_HASH],
```

Comment:
This should now be removed, right?

Reply:
sorry for letting this slip and thanks for pointing out. Have made the relevant changes.